src/sentry/monitors/logic/mark_failed.py

```diff
@@ -22,38 +22,35 @@
 logger = logging.getLogger(__name__)
 
 
-class MonitorFailure:
-    UNKNOWN = "unknown"
-    MISSED_CHECKIN = "missed_checkin"
-    DURATION = "duration"
-
-
 def mark_failed(
-    monitor_env: MonitorEnvironment,
-    last_checkin=None,
-    reason=MonitorFailure.UNKNOWN,
-    occurrence_context=None,
+    failed_checkin: MonitorCheckIn,
+    ts: datetime | None,
 ):
+    monitor_env = failed_checkin.monitor_environment
     failure_issue_threshold = monitor_env.monitor.config.get("failure_issue_threshold", 0)
+
     if failure_issue_threshold:
-        return mark_failed_threshold(failure_issue_threshold, monitor_env, last_checkin)
+        return mark_failed_threshold(failed_checkin, ts, failure_issue_threshold)
     else:
-        return mark_failed_no_threshold(monitor_env, last_checkin, reason, occurrence_context)
+        return mark_failed_no_threshold(failed_checkin, ts)
 
 
 def mark_failed_threshold(
+    failed_checkin: MonitorCheckIn,
+    ts: datetime | None,
     failure_issue_threshold: int,
-    monitor_env: MonitorEnvironment,
-    last_checkin=None,
 ):
     from sentry.signals import monitor_environment_failed
 
+    monitor_env = failed_checkin.monitor_environment
```

Comment:
should we just pass in the `menv` here so we don't pull it out twice?

Reply:
it's fine, it's not going to make multiple queries or anything